test/integration/targets/dnf/tasks/repo.yml

```diff
@@ -0,0 +1,248 @@
+- name: Install foo-1.0-1
+  dnf:
+    name: foo-1.0-1
+    state: present
+  register: dnf_result
+
+- name: Check foo with rpm
+  shell: rpm -q foo
+  register: rpm_result
+
+- name: Verify installation
+  assert:
+    that:
+        - "rpm_result.rc == 0"
+        - "dnf_result.changed"
```

Comment:
Since this test requires the `foo` package to not be installed before the test runs, it would be good to have an `always` block to remove the package at the end of the test. Otherwise re-running the test after a failure part way through could result in this assert failing due to the package already being present.